src/routes/mods.rs

```diff
@@ -85,38 +88,252 @@ fn convert_mod(data: database::models::mod_item::QueryMod) -> models::mods::Mod
     }
 }
 
+/// A mod returned from the API
+#[derive(Serialize, Deserialize)]
+pub struct EditMod {
+    pub title: Option<String>,
+    pub description: Option<String>,
+    pub body: Option<String>,
+    pub status: Option<ModStatus>,
```

Comment:
We should fix those bugs. This way is a lot more cleaner and concise, and reduces repeated code.

Reply:
The issue is that if this is kept within the same route as mod editing, finding those bugs and fixing them becomes significantly harder.  Having a separate moderation route for changing states more closely matches how it *should* be used, makes it more maintainable, and allows for much more careful checking of the code whenever something related to states changes.

I would also recommend instead of allowing users to manually set states, have a route for each allowed transition between states.  Moderators can move mods between processing and approved / rejected; Admins can set a mod's state to anything; users can only move mods from draft to processing and back.  Encoding the allowed states is a better solution because it makes the API easier to use (`PUT /api/v1/mod/{id}/approve` rather than forming a full request with a json body) and acts as a whitelist rather than a blacklist.  If we add more states, they won't be accessible until we specifically allow them to be.  With our current system, if we add a new state without checking and updating this route, normal users will be able to change mods' states to or from the new states without authorization.

Reply:
It's fixed already. And we should keep that in mind when we add a new state. We don't need to "specifically allow them to be", only approved and rejected states are restricted. This method works fine